api.py

```diff
@@ -0,0 +1,127 @@
+
+The OpenModelica software and the Open Source Modelica
+Consortium (OSMC) Public License (OSMC-PL) are obtained
+from OSMC, either from the above address,
+from the URLs: http://www.ida.liu.se/projects/OpenModelica or
+http://www.openmodelica.org, and in the OpenModelica distribution.
+GNU version 3 is obtained from: http://www.gnu.org/copyleft/gpl.html.
+
+This program is distributed WITHOUT ANY WARRANTY; without
+even the implied warranty of  MERCHANTABILITY or FITNESS
+FOR A PARTICULAR PURPOSE, EXCEPT AS EXPRESSLY SET FORTH
+IN THE BY RECIPIENT SELECTED SUBSIDIARY LICENSE CONDITIONS OF OSMC-PL.
+
+See the full OSMC Public License conditions for more details.
+"""
+
+import logging
+from flask_restx import Resource, Api
+import omc_proxy
+import util
+from flask_restx import reqparse
+from flask_restx import inputs
+import svg_writer
+import flask
+import settings
+
+log = logging.getLogger(__name__)
+
+api = Api(version='1.0', title='OMWebService API', description='OMWebService API Documentation')
+
+@api.errorhandler
+def default_error_handler(error):
+  '''Default error handler'''
+  return {'message': str(error)}, getattr(error, 'code', 500)
+
+@api.route("/version")
+class Version(Resource):
+  def get(self):
+    return omc_proxy.sendCommand("getVersion()")
+
+@api.route("/library-nodes/")
+class LibraryNodes(Resource):
+  parser = reqparse.RequestParser()
+  parser.add_argument("name", default = "AllLoadedClasses", help = "A Modelica node name")
+  parser.add_argument("recursive", type = inputs.boolean, default = True, help = "List all node names recursively")
+  parser.add_argument("sort", type = inputs.boolean, default = False, help = "Sort the node names")
+
+  @api.expect(parser)
+  def get(self):
+    try:
+      """Returns the list of library nodes."""
+      args = self.parser.parse_args()
+      nodeName = args["name"]
+      recursive = args["recursive"]
+      sort = util.pythonBoolToModelicaBool(args["sort"])
+
+      nodesJson = []
+      if nodeName == "AllLoadedClasses":
+        classNames = omc_proxy.sendCommand("getClassNames({0}, false, false, {1}, false, true, false)".format(nodeName, sort))
+        for className in classNames:
+          nodesJson.append(self.getNodeJson(className, recursive, False, sort))
+        return nodesJson
+      else:
+        return self.getNodeJson(nodeName, recursive, True, sort)
+      
+    except:
+      raise
+
+  def getNodeJson(self, nodeName, recursive, topLevel, sort):
+    nodeJson = dict()
+    nodeJson["id"] = nodeName
+    nodeJson["displayLabel"] = util.getLastWordAfterDot(nodeName)
+    classInfo = omc_proxy.sendCommand("getClassInformation({0})".format(nodeName))
+    if classInfo[0] == 'package':
+      nodeJson["nodeType"] = "collection"
+    else:
+      nodeJson["nodeType"] = "component"
+
+    (iconGraphics, connectors, parameters) = omc_proxy.getIconGraphicsConnectorsAndParameters(nodeName)
+
+    path = "{0}.svg".format(util.nodeToFileName(nodeName))

```

Comment:
Should this variable be called `fileName` instead of `path`? Same for the filePath variable used elsewhere in this file..